eng/pipelines/templates/generate-insertion.yml

```diff
@@ -27,6 +27,11 @@ jobs:
       # When this isn't provided, the tool sets the --AadAuth flag instead of using --PatAuthEnvVar. The AadAuth doesn't seem to work for this situation.
       AccessToken: $(System.AccessToken)
 
+  # Sets the ShortCommitId variable for the job.
+  # https://docs.microsoft.com/azure/devops/pipelines/process/set-variables-scripts?view=azure-devops&tabs=powershell#set-variable-properties
+  - powershell: Write-Host "##vso[task.setvariable variable=ShortCommitId;isoutput=true]$('$(Build.SourceVersion)'.Substring(0,6))"
```

Comment:
6 characters might not be unique. I've seen collisions on 6 characters of the SHA1. Better to include more here.

See https://twitter.com/drewnoakes/status/830190508727205889 for a 7 character collision I hit in real life. (I think the maths in that tweet is wrong. The chance is 1 in 16^7, or 1 in 268,435,456, or 0.00000037%.)

Reply:
Oy. I meant for this to be 7 characters, but I forgot the parameters are *index*, *length*. 😅 The default for Git is 7 so that's what I was going for. I saw this [response here](https://stackoverflow.com/a/21015031/294804) about length and probabilities and such. I can run `git rev-parse --short` in this small script instead of just truncating the full commit ID (which is `Build.SourceVersion` in an AzDO pipeline). One interesting thing about the idea of using `git rev-parse --short` is that it'll only give an ID that is unique in the *present*; whereas I need to use this ID in the *future* to give a diff over time. Technically, a commit between now and then could conflict with this ID but that is an incredibly low chance of happening.

I think I'll still use the Git command here.

Reply:
The test build resulted in this title, so I still need to fix it anyway. 😂
> Insert DotNet-Project-System (UpdateInsertionPayloadName:20220914.1:$(ShortCommitId)) Payload into main

Reply:
If you can use `git rev-parse --short` then that'd be better. Note that it'd only be unique up until that point. Beyond that, it could conflict. I think just using 10 characters would suffice, no? The odds of a conflict at that point are astronomically low.

Reply:
Latest build with the change to 10 digits worked appropriately.
> Insert DotNet-Project-System (UpdateInsertionPayloadName:20220914.10:a5fea54503) Payload into main